Brick-Studios/BeastArena
# Added level loading and scene manager

# Description

The GameController now contains code which will load a beautifully designed level from a JSON file. I does so by using the JSON adapter from the engine to create a level model. This level model can be converted to entities by the SceneManager. The SceneManager is also able the destroy the whole scene.

At last, some changes were made to the EntityFactory because the engine now contains alpha (transparency).

## Screenshot(s)
![image](https://user-images.githubusercontent.com/21974974/68227479-11f6c980-fff4-11e9-88bf-35cb0af34ed3.png)


# How can this be tested?

You can start off by compiling the game (and thus [engine](https://github.com/Brick-Studios/BrickEngine/pull/39)) and check if the level looks like the screenshot. There are no tests for the SceneManager because that is too complicated but you could tinker around with the JSON level files.

- [ ] The game compiles and looks like the screenshot

# Checklist:
